### DISCUSSION TITLE
Disable mouse-position-scrolling !!

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#2924] [Q&A]
I'm not sure how it's activating but at times when my mouse moves to a position on the screen Sumatra goes into a rolling scroll. It's a horrible experience. I use the mouse wheel, scroll bar, click drag or keyboard to scroll, but never mouse position.
Can someone say how to disable it permanently? I had a glance at the settings but no luck...
Cheers! David
3.5.15239 64-bit win10

**Comment by tactilis** [comment]
Are you accidentally pressing the mouse wheel (which may set to produce a mouse middle-button click)? This is set in some apps to do a 'rolling scroll', the direction and speed of which depends on your mouse position.

Do you get a scrolling symbol on the screen? Possibly like this?

![image](https://user-images.githubusercontent.com/4060092/188273483-447396af-64ce-4a55-b721-770aa5de3750.png)

**Comment by meteorquake** [comment]
Yes that's the sort of thing (trying it I get a cross-arrow, but same effect). It happens a lot, no doubt particularly with the middle button being easy to bump in the mouse position here, which isn't a chair-at-desk setup.
It would be ideal to be able to disable it entirely (as you can in Firefox) as it's more than a nuisance and not in my life will I want to use such mouse-position-scrolling even once :)
Cheers :)

**Comment by tactilis** [comment]
If you just click the middle button (i.e. scroll wheel click) again, it will exit the autoscroll mode.

Here's a possible way to disable middle mouse autoscroll:  https://forum.sumatrapdfreader.org/t/disable-auto-scrolling-with-middle-mouse-button/3424 depending on what mouse and driver you have.

The final option is to take the mouse apart and physically disable scroll wheel click (i.e. stop the switch contacts being closed).

**Reply by GitHubRulesOK** [reply]
@tactilis 
> take the mouse apart

🤣 Beware the cruelty to animals brigade, may bite you
the trad method was superglue a thumbtack point upwards on the escape key. or jamb a matchstick under it

**Comment by GitHubRulesOK** [comment]
Since old forum answers don't always link well here was the past suggestion as linked by @tactilis 

By default, a middle mouse click turns on autoscrolling (scrolling with mouse movement). I would like to disable this behavior. (I want to keep scrolling when I rotate the mouse wheel, just not when I click it.)

Is there any way to do this?


GitHubRulesOK
Middle scrolling is a mouse/pad hardware driver function not controlled by SumatraPDF so to stop it working for example in this browser I need to find the mouse controls and change them.
![image](https://user-images.githubusercontent.com/12754913/188277162-2fc546d8-d304-4abe-ba93-7e6b56911ce1.png)

In some cases (better hardware drivers) you can set the actions per application so you would set a different action when the mouse finds SumatraPDF as its focus.
The simplest way if you are using a budget mouse is to use an AutoHotKey script that targets SumatraPDF thus allowing any mod of keyboard and mouse actions.
I would also recommend for a wide range of customisation https://www.highrez.co.uk/downloads/XMouseButtonControl.htm

**Reply by meteorquake** [reply]
So this naturally begs the question, how does firefox allow the user to enable/disable it, using the about:config boolean option general.autoScroll ?
d
